server/commands/deployments/DeployService.js

```diff
@@ -47,12 +86,37 @@ module.exports = function DeployServiceCommandHandler(command) {
     // Run asynchronously, we don't wait for deploy to finish intentionally
     deploy(deployment, destination, sourcePackage, command);
 
-    let accountName = deployment.accountName;
-    yield deploymentLogger.started(deployment, accountName);
+    yield deploymentLogger.started(deployment, deployment.accountName);
     return deployment;
   });
 };
 
+function checkAsgSchedule(asg) {
+  let state = getScheduleStateByAsg(asg);
+  if (state && state.toLowerCase() === 'off') {
+    throw new Error('The state of this ASG schedule is off.');
+  }
+}
+
+function checkEnvironmentSchedule(environment) {
+  let environmentSchedule = getScheduleByEnvironment(environment);
+  let environmentScheduleState;
+
+  if (environmentSchedule && environmentSchedule.parseResult) {
+    if (Array.isArray(environmentSchedule.parseResult.schedule)) {
+      environmentScheduleState = getExpectedState.fromMultipleEnvironmentSchedules(environmentSchedule.parseResult.schedule);
+    } else {
+      environmentScheduleState = getExpectedState.fromSingleEnvironmentSchedule(environmentSchedule.parseResult);
+    }
+  }
+
+  if (environmentScheduleState) {
+    if (environmentScheduleState.toLowerCase() === 'off') {
+      throw new Error('The state for the environment schedule is off.');
```

Comment:
Could we make this less ambiguous and just _"The Environment you are trying to deploy to is currently scheduled Off"_